FreeBSD building broken by #19438

All building is broken by #19438 with a compiler error. Caused by commit https://github.com/mono/mono/commit/429d269d0f83e6b382b8d36930a5fdb2b066b618

```
mini-posix.c:694:19: error: incompatible pointer types passing 'clockid_t *' (aka 'int *') to parameter of type 'mono_clock_id_t *' (aka 'void **') [-Werror,-Wincompatible-pointer-types]
```

--- a/src/mono/configure.ac
+++ b/src/mono/configure.ac
@@ -761,11 +761,6 @@ if test "x$host_linux" = "xyes" -a x$ac_cv_header_sys_mman_h = xyes -a  x$ac_cv_
 	AC_DEFINE(ENABLE_JIT_DUMP, 1, [Enable jit dump support on Linux])
 fi
 
-# for mono/utils/mono-time.h
-AC_CHECK_TYPES([clockid_t], [AC_DEFINE(HAVE_CLOCKID_T)], [],
-	[#include <sys/types.h>])
-
-
 # for support
 AC_CHECK_HEADERS(poll.h)
 AC_CHECK_HEADERS(sys/poll.h)
@@ -2212,6 +2207,9 @@ if test x$host_win32 = xno; then
 	AC_CHECK_FUNC(mach_timebase_info, [AC_DEFINE(HAVE_MACH_TIMEBASE_INFO, 1, [mach_timebase_info])])
 	AC_CHECK_FUNC(futimes,            [AC_DEFINE(HAVE_FUTIMES, 1, [futimes])])
 	AC_CHECK_FUNC(futimens,           [AC_DEFINE(HAVE_FUTIMENS, 1, [futimens])])
+
+	AC_CHECK_TYPES([clockid_t], [AC_DEFINE(HAVE_CLOCKID_T)], [],
+		[#include <sys/types.h>])
 
 	dnl hires monotonic clock support
 	AC_SEARCH_LIBS(clock_gettime, rt)

--- a/src/mono/configure.ac
+++ b/src/mono/configure.ac
@@ -758,8 +758,13 @@ AC_CHECK_HEADERS(elf.h)
 # for perf jit dump support
 AC_CHECK_HEADERS(sys/mman.h)
 if test "x$host_linux" = "xyes" -a x$ac_cv_header_sys_mman_h = xyes -a  x$ac_cv_header_elf_h = xyes -a x$ac_cv_header_sys_syscall_h = xyes; then
-   AC_DEFINE(ENABLE_JIT_DUMP, 1, [Enable jit dump support on Linux])
+	AC_DEFINE(ENABLE_JIT_DUMP, 1, [Enable jit dump support on Linux])
 fi
+
+# for mono/utils/mono-time.h
+AC_CHECK_TYPES([clockid_t], [AC_DEFINE(HAVE_CLOCKID_T)], [],
+	[#include <sys/types.h>])
+
 
 # for support
 AC_CHECK_HEADERS(poll.h)

--- a/src/mono/mono/utils/mono-time.h
+++ b/src/mono/mono/utils/mono-time.h
@@ -32,7 +32,7 @@ gint64 mono_100ns_datetime_from_timeval (struct timeval tv);
 #if defined(HOST_DARWIN)
 #include <mach/clock.h>
 typedef clock_serv_t mono_clock_id_t;
-#elif defined(HAVE_SYS_TIME_H)
+#elif defined(HAVE_CLOCKID_T)
 typedef clockid_t mono_clock_id_t;
 #else
 typedef void *mono_clock_id_t;

--- a/src/mono/mono/utils/mono-time.h
+++ b/src/mono/mono/utils/mono-time.h
@@ -32,8 +32,7 @@ gint64 mono_100ns_datetime_from_timeval (struct timeval tv);
 #if defined(HOST_DARWIN)
 #include <mach/clock.h>
 typedef clock_serv_t mono_clock_id_t;
-#elif defined(__unix__)
-#include <sys/types.h>
+#elif defined(HAVE_SYS_TIME_H)
 typedef clockid_t mono_clock_id_t;
 #else
 typedef void *mono_clock_id_t;

--- a/src/mono/mono/utils/mono-time.h
+++ b/src/mono/mono/utils/mono-time.h
@@ -32,7 +32,7 @@ gint64 mono_100ns_datetime_from_timeval (struct timeval tv);
 #if defined(HOST_DARWIN)
 #include <mach/clock.h>
 typedef clock_serv_t mono_clock_id_t;
-#elif defined(__linux__)
+#elif defined(__unix__)
 #include <sys/types.h>
 typedef clockid_t mono_clock_id_t;
 #else

